File: org.kevoree.modeling.microframework/src/main/java/org/kevoree/modeling/api/data/cache/DimensionCache.java
package org.kevoree.modeling.api.data.cache;

import org.kevoree.modeling.api.KDimension;
import org.kevoree.modeling.api.ModelListener;
import org.kevoree.modeling.api.time.DefaultTimeTree;
import org.kevoree.modeling.api.time.TimeTree;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

/**
 * Created by duke on 10/30/14.
 */
public class DimensionCache {
    public Map<Long, TimeTree> timeTreeCache = new HashMap<Long, TimeTree>();
    public Map<Long, TimeCache> timesCaches = new HashMap<Long, TimeCache>();
    public KDimension dimension;
    public TimeTree rootTimeTree = new DefaultTimeTree();
    public List<ModelListener> listeners = new ArrayList<ModelListener>();

    public DimensionCache(KDimension dimension) {
        this.dimension = dimension;
    }
}

File: org.kevoree.modeling.microframework.typescript/src/main/ts/org/kevoree/modeling/api/data/cache/DimensionCache.ts
///<reference path="../../KDimension.ts"/>
///<reference path="../../ModelListener.ts"/>
///<reference path="../../time/DefaultTimeTree.ts"/>
///<reference path="../../time/TimeTree.ts"/>
///<reference path="../../../../../../java/util/JUArrayList.ts"/>
///<reference path="../../../../../../java/util/JUHashMap.ts"/>
///<reference path="../../../../../../java/util/JUList.ts"/>
///<reference path="../../../../../../java/util/JUMap.ts"/>

class DimensionCache {

  public timeTreeCache: JUMap<number, TimeTree> = new JUHashMap<number, TimeTree>();
  public timesCaches: JUMap<number, TimeCache> = new JUHashMap<number, TimeCache>();
  public dimension: KDimension<any,any,any> = null;
  public rootTimeTree: TimeTree = new DefaultTimeTree();
  public listeners: JUList<ModelListener> = new JUArrayList<ModelListener>();

  constructor(dimension: KDimension<any,any,any>) {
    this.dimension = dimension;
  }

}

